src/storhaug

```diff
@@ -358,13 +364,23 @@ setup_create_resources()
             options="_netdev,defaults,direct-io-mode=enable,transport=tcp,xlator-option=*client*.ping-timeout=10" \
         --clone ganesha_state-clone ganesha_state meta interleave="true" clone-max="${STORAGE_NUM_SERVERS}"
 
+    mkdir -p "${HA_MNT_DIR}/${HA_SMB_MNT_DIR}"
+    pcs -f ${cibfile} resource create ctdb_lock ocf:heartbeat:Filesystem \
+        params \
+            device="localhost:/${HA_SMB_VOL}" \
+            directory="${HA_MNT_DIR}/${HA_SMB_MNT_DIR}" \
+            fstype="glusterfs" \
+            options="_netdev,defaults,direct-io-mode=enable,transport=tcp,xlator-option=*client*.ping-timeout=10" \
+        --clone ctdb_lock-clone ctdb_lock meta interleave="true" clone-max="${STORAGE_NUM_SERVERS}"
+
     pcs -f ${cibfile} constraint location ganesha_state-clone rule resource-discovery=exclusive score=0 role eq storage
 
     pcs cluster cib-push ${cibfile} || storlog "ERR" "Failed to create filesystem resources."
 
     # CTDB
     pcs -f ${cibfile} resource create ctdb ocf:heartbeat:CTDB \
         params \
+            ctdb_recovery_lock="${HA_MNT_DIR}/${HA_SMB_MNT_DIR}/ctdb.lock" \
```

Comment:
@guihecheng ...oh right! Seems I forgot to submit my work to the right upstream repository. If you look at `vagrant-ansible/playbooks/files/CTDB` I've done a lot of work to the CTDB RA to get it to leverage more of Pacemaker's functionality. And in the ansible installation I copy it into the right place on the VM to work around the upstream... but of course, this doesn't help in real world scenarios.

I'll try to submit my changes this week. In the mean time, you should be able to just copy the CTDB file to `/usr/lib/ocf/resource.d/heartbeat/CTDB` and it should work. Let me know if it does or doesn't.